tests/Tests/Properties/Folds.hs

```diff
@@ -190,6 +194,13 @@ tl_unfoldrN n m   = (L.take i . L.unfoldr (unf j)) `eq`
     where i = fromIntegral (n :: Word16)
           j = fromIntegral (m :: Word16)
 
+isAscii_border :: IO ()
+isAscii_border = do
+    -- ASCII prefix ends at position 3 (from 0)
+    let text  = T.pack "123一二三"
+        text' = case text of T.Text arr off _len -> T.Text arr (off+1) 3
+    assertBool "UTF-8 string with ASCII prefix ending at last position incorrectly detected as ASCII" $ not $ T.isAscii text'
```

Comment:
Instead of pattern-matching and constructing an invalid `Text`, you can use `T.drop` to safely get a non-zero offset. And then the ascii prefix should be longer than the byte-length after `T.drop` to properly test against accidentally ignoring the offset.

Reply:
thanks, made those changes